x-pack/plugins/drilldowns/url_drilldown/public/lib/url_drilldown.tsx

```diff
@@ -109,18 +111,37 @@ export class UrlDrilldown implements Drilldown<Config, UrlTrigger, ActionFactory
       console.warn(
         `UrlDrilldown [${config.url.template}] is not valid. Error [${error}]. Skipping execution.`
       );
+      return false;
     }
 
-    return Promise.resolve(isValid);
+    const url = this.buildUrl(config, context);
+    const validUrl = this.deps.externalUrl.validateUrl(url);
+    if (!validUrl) {
+      return false;
+    }
+
+    return true;
   };
 
-  public readonly getHref = async (config: Config, context: ActionContext) => {
-    const scope = this.getRuntimeVariables(context);
-    return urlDrilldownCompileUrl(config.url.template, scope);
+  private buildUrl(config: Config, context: ActionContext): string {
```

Comment:
nit: do we really need this function ? looks like unnecesarry complication

Reply:
Yes, I've put this code into this separate function because this code is now used in two places in this class. Also, if I didn't add `.getHref()` into `.execute()` in this PR this code would have been used in 3 places in this class.